### ISSUE TITLE
How do I use sample_weight with an existing train-valid split?

### ISSUE THREAD

**Issue by gloryVine** [skorch-dev/skorch#801]
The [documentation](https://skorch.readthedocs.io/en/stable/user/FAQ.html) explains how to use `sample_weight`, and how to use existing train and validation sets, but it doesn't explain how I can use both at the same time.

Specifically, I have the following LGBM code:
```
train_weights = ...
val_weights = ...
self.fitting_parameters = dict(sample_weight=train_weights,
                                       eval_set=(X_valid, y_valid),
                                       eval_sample_weight=[val_weights],
                                       eval_metric=my_metric(),
                                       verbose=-1)
```
which I want to adapt to a skorch neural network.

At the moment I do (sketch)
```
        self.train_dataset = ...
        self.valid_dataset = ...
        self.fitting_parameters = None

        self.net = skorch.NeuralNetRegressor(module=FFNNModule(
            train_split=predefined_split(self.valid_dataset))
        self.net.fit(self.train_dataset, y=None)
```

How can I pass`train_weights` and `valid_weights` so that they are used for the `self.train_dataset` respectively `self.valid_dataset` in the correct way when computing the loss?

**Comment by BenjaminBossan** [comment]
So do I understand correctly that you followed [this suggestion](https://skorch.readthedocs.io/en/stable/user/FAQ.html#i-want-to-use-sample-weight-how-can-i-do-this) to add support for sample weights during training? So your `X_train` (before passing it to the dataset) is a dict with an additional `sample_weight` key? If you did that, the model should be trained correctly, taking sample weights into account.

For your validation data, sample weights only matter in the sense that the validation loss might look different. Here, do the exact same thing for `X_valid` as you did for `X_train` and I think the results should be correct, i.e. when `get_loss` is called on your validation data, sample weights are taken into account. Therefore, in the report you see and in the `net.history`, `valid_loss` correctly includes the sample weights.

If you would like additional metrics (e.g. added via `EpochScoring` callbacks) to also take into account the sample weights, you'd need to do some extra work to achieve this.

**Comment by BenjaminBossan** [comment]
Since there was no reply for some time now, I consider the question answered. If not, feel free to re-open.
